go/logic/inspect.go

```diff
@@ -195,12 +196,22 @@ func (this *Inspector) validateConnection() error {
 	if len(this.connectionConfig.Password) > mysql.MaxReplicationPasswordLength {
 		return fmt.Errorf("MySQL replication length limited to 32 characters. See https://dev.mysql.com/doc/refman/5.7/en/assigning-passwords.html")
 	}
-	query := `select @@global.port, @@global.version`
+
+	extraPortRegexp := regexp.MustCompile("Percona|Maria")
+	query := `select @@global.port, @@global.version, @@global.version_comment`
 	var port int
-	if err := this.db.QueryRow(query).Scan(&port, &this.migrationContext.InspectorMySQLVersion); err != nil {
+	var extraPort int
+	var versionComment string
+	if err := this.db.QueryRow(query).Scan(&port, &this.migrationContext.InspectorMySQLVersion, &versionComment); err != nil {
 		return err
 	}
-	if port != this.connectionConfig.Key.Port {
+	if extraPortRegexp.MatchString(versionComment) {
+		extraPortQuery := `select @@global.extra_port`
+		if err := this.db.QueryRow(extraPortQuery).Scan(&extraPort); err != nil {
+			return err
+		}
+	}
+	if port != this.connectionConfig.Key.Port && extraPort != this.connectionConfig.Key.Port {
 		return fmt.Errorf("Unexpected database port reported: %+v", port)
```

Comment:
I do wonder if this error message should include the extra port (if it's available). Not sure.

Reply:
Yes. Do let's `return log.Errorf("problem reading @@global.extra_port: %+v", err)`